pytensor/compile/builders.py

```diff
@@ -47,11 +48,15 @@ def infer_shape(outs, inputs, input_shapes):
             assert len(inp_shp) == inp.type.ndim
 
     shape_feature = ShapeFeature()
-    shape_feature.on_attach(FunctionGraph([], []))
+    fgraph = FunctionGraph([], [], features=[shape_feature])
+    for v in chain.from_iterable(s for s in input_shapes if s is not None):
+        # Import input_shape nodes, as for some graphs ShapeFeature assumes these were seen before
+        if (node := v.owner) is not None:
+            fgraph.import_node(node, import_missing=True)
```

Comment:
ShapeFeature is quite picky about having seen some variables. It may even count as bug but I don't want to open that can of worms.

Anyway OpFromGraph uses this helper for infer_shape, and the way Blockwise triggers it when it has an OFG inside, would end up triggering an error.